src/miner.cpp

```diff
@@ -173,17 +173,18 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
     return std::move(pblocktemplate);
 }
 
-void BlockAssembler::onlyUnconfirmed(CTxMemPool::setEntries& testSet)
+void BlockAssembler::onlyUnconfirmed(CTxMemPool::vecEntries& testSet)
 {
-    for (CTxMemPool::setEntries::iterator iit = testSet.begin(); iit != testSet.end(); ) {
-        // Only test txs not already in the block
-        if (inBlock.count(*iit)) {
-            testSet.erase(iit++);
-        }
-        else {
-            iit++;
-        }
+    size_t max_element = testSet.size(), i = 0;
+    while (i < max_element) {
+        bool not_in_block = !inBlock.count(testSet[i]);
+        // if !not_in_block, set testSet[i] to testSet[max_element-1]
+        max_element -= !not_in_block;
+        testSet[i] = testSet[!not_in_block*max_element + not_in_block*i];
+        i += not_in_block;
     }
+    // drop erased elements
+    testSet.resize(max_element);
```

Comment:
This can be replaced by a std::remove_if and a std::erase idiom, which will be of similar behavior (remove_if keeps order stable) and same complexity. Much easier to read and less chance of bugs.